src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskIntegrationTest.java

```diff
@@ -305,7 +329,7 @@ void testSourceCanHandleNonExistentDatabaseAndSurviveDroppingWithPipeline() {
 
       db.drop();
       assertNull(task.poll());
-
+      assertNull(task.poll());
```

Comment:
Extra poll required to reconstruct the cursor, due to the simplified cursor logic.